extensions/sql/contract-definition/loader/src/main/java/org/eclipse/dataspaceconnector/sql/contractdefinition/loader/SqlContractDefinitionLoader.java

```diff
@@ -0,0 +1,118 @@
+/*
+ *  Copyright (c) 2022 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.sql.contractdefinition.loader;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.dataspaceconnector.dataloading.ContractDefinitionLoader;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.persistence.EdcPersistenceException;
+import org.eclipse.dataspaceconnector.spi.transaction.TransactionContext;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractDefinition;
+import org.eclipse.dataspaceconnector.sql.contractdefinition.schema.SqlContractDefinitionTables;
+
+import java.sql.Connection;
+import java.util.Objects;
+import javax.sql.DataSource;
+
+import static org.eclipse.dataspaceconnector.sql.SqlQueryExecutor.executeQuery;
+
+public class SqlContractDefinitionLoader implements ContractDefinitionLoader {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
```

Comment:
should be passed in through the constructor since we're using a centrally configured OM

Reply:
Since the central ObjectMapper might be configured to serialize data in a way to be incompatible to what's expected to be written/read into/from database, I'd argue for have a dedicated **private** ObjectMapper instance here. Also the ContractDefinitionLoader/Store abstracts away the burden to take care of how data is serialized.

Reply:
can you give an example of such an incompatibility? the central OR mapper uses `@JsonType`s for example. 

Apart from that i don't see any sense in serializing objects such as contract definitions to JSON, only to store them in a relational database. The discussion is therefore moot and the ObjectMapper should go away altogether.

Reply:
The TypeManager is designed specifically for the case of polymorphic deserialization in the runtime and *must* be used by extensions for this mechanism to work. Please do not use a private ObjectMapper.